packages/x-tree-view/src/internals/plugins/useTreeViewFocus/useTreeViewFocus.ts

```diff
@@ -3,38 +3,73 @@ import useEventCallback from '@mui/utils/useEventCallback';
 import { EventHandlers } from '@mui/base/utils';
 import ownerDocument from '@mui/utils/ownerDocument';
 import { TreeViewPlugin } from '../../models';
-import { populateInstance } from '../../useTreeView/useTreeView.utils';
+import { populateInstance, populatePublicAPI } from '../../useTreeView/useTreeView.utils';
 import { UseTreeViewFocusSignature } from './useTreeViewFocus.types';
 import { useInstanceEventHandler } from '../../hooks/useInstanceEventHandler';
 
 export const useTreeViewFocus: TreeViewPlugin<UseTreeViewFocusSignature> = ({
   instance,
+  publicAPI,
   params,
   state,
   setState,
   models,
   rootRef,
 }) => {
+  const focusedNodeId = state.focusedNodeId;
   const setFocusedNodeId = useEventCallback((nodeId: React.SetStateAction<string | null>) => {
-    const cleanNodeId = typeof nodeId === 'function' ? nodeId(state.focusedNodeId) : nodeId;
-    setState((prevState) => ({ ...prevState, focusedNodeId: cleanNodeId }));
+    const cleanNodeId = typeof nodeId === 'function' ? nodeId(focusedNodeId) : nodeId;
+    if (focusedNodeId !== cleanNodeId) {
+      setState((prevState) => ({ ...prevState, focusedNodeId: cleanNodeId }));
+    }
   });
 
+  const isTreeFocused = React.useCallback(
```

Comment:
You can find a more robust implementation in #12226 
It will support shadow DOM this way :ok_hand: 

If you want you can replace the `contains` with an equality like on yours, I will change for a `contain` when my PR get merged. The important point is `getActiveElement` which is more precise than `document.activeElement`